src/foam/lib/query/StringParser.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.lib.query;
+
+import foam.lib.json.ASCIIEscapeParser;
+import foam.lib.json.UnicodeParser;
+import foam.lib.parse.PStream;
+import foam.lib.parse.Parser;
+import foam.lib.parse.ParserContext;
+
+public class StringParser implements Parser {
+  public final static char ESCAPE = '\\';
+
+  public StringParser() {
+  }
+
+  public PStream parse(PStream ps, ParserContext x) {
+
+    if ( !ps.valid() )  return null;

```

Comment:
I think our purposes can be accomplished by just parsing any character that isn't whitespace.

We're only using this string parser to parse values now, not to parse property named.

I don't see why beforeHead() would be a userful delimiter, if beforeHead was a ":" that doesn't mean the string is going to end with a colon.

Just capturing characters until the next whitespace or close parenthesis should be sufficient.